src/enzo/Grid_SchrodingerSolver.C

```diff
@@ -0,0 +1,350 @@
+c     gr_x,y,zacc - gravitational acceleration fields
+c     gravity - flag indicating whether or not to use gravity field (1 = yes)
+c     gridvel - bulk grid velocity (vector of length 3)
+c     i,j,kn  - dimensions of field arrays
+c     idiff   - diffusion flag (0 = off)
+c     idual   - dual energy formalism flag (0 = off)
+c     ipresfree - pressure free flag (0 = off, 1 = on, i.e. p=0)
+c     nhy     - cycle number (for better operator splitting)
+c     rank    - dimension of problem (not currently used)
+c     start   - array (of dimension 3) specifying the start of the active
+c               region fo reach dimension (zero based)
+c     tmp     - temporary work space (30 * largest_slice)
+c     u       - x-velocity field
+c     v       - y-velocity field
+c     w       - z-velocity field
+c     bottom  - true (1) if this is the lowest level
+c     minsupecoef - coefficient for minimum pressure support (0 - not used)
+/
+************************************************************************/
+
+// Solve the hydro equations with the solver, saving the subgrid fluxes
+//
+
+#include <stdio.h>
+#include <math.h>
+#include "ErrorExceptions.h"
+#include "macros_and_parameters.h"
+#include "typedefs.h"
+#include "global_data.h"
+#include "Fluxes.h"
+#include "GridList.h"
+#include "ExternalBoundary.h"
+#include "Grid.h"
+#include "fortran.def"
+
+int rk3tvd(float *repsi, float *impsi, float *repsi0, float *impsi0,
+         float alpha, float beta,
+	       int in, int jn, int kn, int rank,
+	       int is, int ie, int js, int je, int ks, int ke, 
+	       float dt, float dx[], float dy[], float dz[],
+	       float hmcoef,
+         int gravity, double *p);
+
+int rk4(double *repsi, double *impsi,
+         int in, int jn, int kn, int rank,
+         double dt, double dx[], double dy[], double dz[],
+         double hmcoef);
+
+int sv2(int nhy, double *repsi, double *impsi,
+         int in, int jn, int kn, int rank,
+         double dt, double dx[], double dy[], double dz[],
+         double hmcoef);
+
+int SchrdingerAddPotential(double *repsi, double *impsi,
+         int in, int jn, int kn, int rank,
+         int gin, int gjn, int gkn,
+         double dt, 
+         double hmcoef,
+         double *p, int start1, int start2, int start3);
+
+int GetUnits (float *DensityUnits, float *LengthUnits,
+         float *TemperatureUnits, float *TimeUnits,
+         float *VelocityUnits, double *MassUnits, FLOAT Time);
+int FindField(int field, int farray[], int numfields);
+int QuantumGetUnits(float *DensityUnits, float *LengthUnits,
+       float *TemperatureUnits, float *TimeUnits,
+       float *VelocityUnits, FLOAT Time);
+int CosmologyComputeExpansionFactor(FLOAT time, FLOAT *a, FLOAT *dadt);
+
+
+int grid::SchrodingerSolver( int nhy )
+{
+  /*  Locals */
+  if (ProcessorNumber != MyProcessorNumber)
+    return SUCCESS;
+
+  int i, ie, is, j, je, js, k, ks, ke, n, ixyz, ret;
+  //int DensNum, GENum, TENum, Vel1Num, Vel2Num, Vel3Num;
```

Comment:
lots of commented-out lines in this file - can they be removed?

Reply:
removed